pkg/endpoint/bpf.go

```diff
@@ -1236,16 +1247,17 @@ func (e *Endpoint) applyPolicyMapChanges() error {
 	}
 
 	if errors > 0 {
-		return fmt.Errorf("updating desired PolicyMap state failed")
+		return true, fmt.Errorf("updating desired PolicyMap state failed")
```

Comment:
(github didn't allow to comment on un-changed code, so I just add the comment here)

Does the above block guarded `e.visibilityPolicy != nil`  change policy map as well? This comment is not a blocker as visibility annotation feature is considered as deprecated.

Reply:
> Does the above block guarded e.visibilityPolicy != nil change policy map as well?

Great question! (and thanks for asking it). Yes, `e.desiredPolicy.GetPolicyMap().AddVisibilityKeys(e, redirectPort, visMeta, changes)` adds new entries to `adds`. So, we have to check `len(adds)` at the end. (I had a version of this code where I got this wrong, heh).